resources/fetching/fetchers/azure/assets_fetcher.go

```diff
@@ -78,25 +90,33 @@ func NewAzureAssetsFetcher(log *logp.Logger, ch chan fetching.ResourceInfo, prov
 func (f *AzureAssetsFetcher) Fetch(ctx context.Context, cMetadata fetching.CycleMetadata) error {
 	f.log.Info("Starting AzureAssetsFetcher.Fetch")
 	// This might be relevant if we'd like to fetch assets in parallel in order to evaluate a rule that uses multiple resources
-	assets, err := f.provider.ListAllAssetTypesByName(ctx, maps.Keys(AzureAssetTypeToTypePair))
-	if err != nil {
-		return err
+	assetsByAssetGroup := make(map[string][]inventory.AzureAsset, len(AzureAssetGroups))
+	for _, assetGroup := range maps.Keys(AzureAssetGroups) {
```

Comment:
I feel adding another `for` level is confusing. I suggest keeping the old `Fetch` function as `fetchAssetGroup()` parametrized by the asset group name and the new `Fetch` function being just:
```
for _, assetGroup := range range maps.Keys(AzureAssetGroups {
    err = f.fetchAssetGroup(ctx, cMetadata, assetGroup)
    if err ...
}
```